web-common/src/features/entity-management/WatchResourcesClient.ts

```diff
@@ -0,0 +1,249 @@
+import { fileArtifacts } from "@rilldata/web-common/features/entity-management/file-artifacts";
+import { refreshResource } from "@rilldata/web-common/features/entity-management/resource-invalidations";
+import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+import {
+  getConnectorServiceOLAPListTablesQueryKey,
+  getRuntimeServiceGetResourceQueryKey,
+  getRuntimeServiceListResourcesQueryKey,
+  V1ReconcileStatus,
+  V1Resource,
+  V1ResourceEvent,
+  V1WatchResourcesResponse,
+} from "@rilldata/web-common/runtime-client";
+import {
+  invalidateChartData,
+  invalidateMetricsViewData,
+  invalidateProfilingQueries,
+  invalidationForMetricsViewData,
+} from "@rilldata/web-common/runtime-client/invalidation";
+import { isProfilingQuery } from "@rilldata/web-common/runtime-client/query-matcher";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import { WatchRequestClient } from "@rilldata/web-common/runtime-client/watch-request-client";
+import { get } from "svelte/store";
+
+const MainResourceKinds: {
+  [kind in ResourceKind]?: true;
+} = {
+  [ResourceKind.Source]: true,
+  [ResourceKind.Model]: true,
+  [ResourceKind.MetricsView]: true,
+  [ResourceKind.Component]: true,
+  [ResourceKind.Dashboard]: true,
+};
+const UsedResourceKinds: {
+  [kind in ResourceKind]?: true;
+} = {
+  [ResourceKind.ProjectParser]: true,
+  [ResourceKind.Theme]: true,
+  ...MainResourceKinds,
+};
+
+export class WatchResourcesClient {
```

Comment:
**Issue**: The class WatchResourcesClient is exported but does not encapsulate its instantiation logic fully, which could lead to improper usage or instantiation in an undesired state. <br> **Fix**: Encapsulate the instantiation of WatchResourcesClient by making the constructor private and handling all initialization logic within the static New() method to ensure the class is always in a valid state when used. <br> **Code Suggestion**: 
 ```
 - public constructor() {
 + private constructor() {
 ```